fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildPropertyDeriver.java

```diff
@@ -468,13 +482,39 @@ private HashDistributionSpec createLocalByByHashColumns(List<Integer> hashColumn
         return Optional.of(requireDistributionDesc);
     }
 
+    private Void tryGatherForBroadcastJoin(PhysicalHashJoinOperator node, ExpressionContext context) {
+        List<Pair<PhysicalPropertySet, List<PhysicalPropertySet>>> result = Lists.newArrayList();
+        if ((context.getChildOperator(0) instanceof LogicalAggregationOperator)) {
+            LogicalAggregationOperator childOperator = (LogicalAggregationOperator) context.getChildOperator(0);
```

Comment:
I think you should check instance count, not check aggregate operator explicitly